psi4/src/psi4/dfocc/ccd_W_intr.cc

```diff
@@ -154,6 +154,124 @@ void DFOCC::ccd_WmnijT2() {
 
 }  // end ccd_WmnijT2
 
+//======================================================================
+//    WmnijT2AA
+//======================================================================
+void DFOCC::ccd_W_MNIJT2AA()
+{
+    SharedTensor2d J, W, I, X, Y, T;
+    SharedTensor2d T2, Tau, T2new;
+
+    // W_MNIJ Alpha Block
+    // W_MNIJ = <MN||IJ> + \sum_(EF) T(IJ,EF) * <MN|EF>
+    // W_MNIJ = <MN||IJ>
+    J = std::make_shared<Tensor2d>("J (IM|JN)", naoccA, naoccA, naoccA, naoccA);
+    J->gemm(true, false, bQijA, bQijA, 1.0, 0.0);
+    W = std::make_shared<Tensor2d>("W <MN|IJ>", naoccA, naoccA, naoccA, naoccA);
+    W->sort(1324, J, 1.0, 0.0);
+    W->sort(1342, J, -1.0, 1.0);
+    J.reset();
+
+    //W_MNIJ += \sum_(EF) T(IJ,EF) * <MN|EF>
+    Tau = std::make_shared<Tensor2d>("T2 <IJ|AB>", naoccA, naoccA, navirA, navirA);
+    Tau->read_anti_symm(psio_, PSIF_DFOCC_AMPS);
+    J = std::make_shared<Tensor2d>("J (ME|NF)", naoccA, navirA, naoccA, navirA);
+    J->gemm(true, false, bQiaA, bQiaA, 1.0, 0.0);
+    I = std::make_shared<Tensor2d>("I <MN|EF>", naoccA, naoccA, navirA, navirA);
+    I->sort(1324, J, 1.0, 0.0);
+    J.reset();
+    W->gemm(false, true, I, Tau, 1.0, 1.0);
+    I.reset();
+
+    // 0.5 * Tau * WMNIJ
+    T2new = std::make_shared<Tensor2d>("New T2 <IJ|AB>", naoccA, naoccA, navirA, navirA);
+    T2new->read_anti_symm(psio_, PSIF_DFOCC_AMPS);
+    T2new->gemm(true, false, W, Tau, 0.5, 1.0);
+    W.reset();
+    Tau.reset();
+    T2new->write_anti_symm(psio_, PSIF_DFOCC_AMPS);
+    T2new.reset();
```

Comment:
```suggestion
```
Shared pointers are cleared immediately upon leaving scope: this serves no purpose.